cmd/run/run.go

```diff
@@ -706,8 +714,9 @@ func (s *ServerContext) Run(ctx context.Context, config *serverconfig.Config) er
 
 	datastore.Close()
 
-	_ = tp.ForceFlush(ctx)
-	_ = tp.Shutdown(ctx)
+	if tracerProviderCloser != nil {
+		tracerProviderCloser()
```

Comment:
It blows my mind that `NoOpTracerProvider` doesn't provide a NoOp `Shutdown` method.  It feels like we _should_ be able to do 

```
         tp := trace.NewNoopTracerProvider()
	if config.Trace.Enabled {
		// ... 
		tp = telemetry.MustNewTracerProvider(options...)
	}

	otel.SetTracerProvider(tp)

        // ...

       tp.Shutdown()
       tp.ForceFlush(ctx)
```

Are we doing this properly? I briefly checked https://github.com/open-telemetry/opentelemetry-go/issues?q=is%3Aissue+is%3Aopen+nooptracerprovider but no luck...

Reply:
Yeah, they originally had a more normalized noop package but they deprecated it AFAIK.